node-cli/users.js

```diff
@@ -0,0 +1,106 @@
+const inquirer = require("inquirer");
+const Realm = require("realm");
+const index = require("./index");
+const config = require("./config");
+const main = require("./main");
+const output = require("./output");
+
+const REALM_APP_ID = config.realmAppId;
+const appConfig = {
+  id: REALM_APP_ID,
+  timeout: 10000,
+};
+
+const app = new Realm.App(appConfig);
+
+authedUser = {};
+
+async function logIn() {
+  const input = await inquirer.prompt([
+    {
+      type: "input",
+      name: "email",
+      message: "Email:",
+    },
+    {
+      type: "password",
+      name: "password",
+      message: "Password:",
+      mask: "*",
+    },
+  ]);
+
+  try {
+    const credentials = Realm.Credentials.emailPassword(
+      input.email,
+      input.password
+    );
+    const user = await app.logIn(credentials);
+    if (user) {
+      authedUser = user;
+      output.result("You have successfully logged in as " + authedUser.id);
+      return main.mainMenu();
+    } else {
+      output.error("There was an error logging you in");
+      return logIn();
+    }
+  } catch (err) {
+    output.error(JSON.stringify(err, " ", 3));
+    return logIn();
+  }
+}
+
+async function registerUser() {
+  output.header("WELCOME, NEW USER");
+  const input = await inquirer.prompt([
+    {
+      type: "input",
+      name: "email",
+      message: "Email:",
+    },
+    {
+      type: "password",
+      name: "password",
+      message: "Password:",
+      mask: "*",
+    },
+  ]);
+
+  try {
+    const result = await app.emailPasswordAuth.registerUser(
+      input.email,
+      input.password
+    );
+    output.result(result);
```

Comment:
`registerUser()` resolves to `void` so there won't ever actually be a value for `result`.